x/cheqd/types/did_validation.go

```diff
@@ -0,0 +1,220 @@
+package types
+
+import (
+	"github.com/cheqd/cheqd-node/x/cheqd/utils"
+	"regexp"
+)
+
+var SplitDIDRegexp, _        = regexp.Compile(`did:([^:]+?)(:([^:]+?))?:([^:]+)$`)
+var DidNamespaceRegexp, _    = regexp.Compile(`^[a-zA-Z0-9]$`)
+// Base58 only allowed (without OolI and 0)
+var UniqueIDRegexp, _        = regexp.Compile(`^[123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz]*$`)
+// That for groups:
+// Example: did:cheqd:testnet:fafdsffq11213343/path-to-s/ome-external-resource?query#key1???#123
```

Comment:
Is this DID valid? Can fragment be `#key1???#123`?

Reply:
no, it's not valid. I'll change it.